commit 92df8e9c2e021ffa079b4593c83880a81b699746
Author: Thomas Adam <thomas@xteddy.org>
Date:   Thu Aug 23 20:15:49 2012 +0100

    Use correct counter for choose-tree
    
    When displaying items in choose-tree, and when we're only showing windows,
    ensure we use the correct counter.

diff --git a/cmd-choose-tree.c b/cmd-choose-tree.c
index 7132017..94698e7 100644
--- a/cmd-choose-tree.c
+++ b/cmd-choose-tree.c
@@ -203,7 +203,8 @@ windows_only:
 
 			window_choose_add_window(wl->window->active,
 				ctx, s2, wm, final_win_template,
-				final_win_action, idx_ses);
+				final_win_action,
+				(wflag && !sflag) ? win_ses : idx_ses);
 
 			free(final_win_action);
 		}

commit 1139192fc04a6d61dc432bd8b1174ae47e98b0eb
Author: Thomas Adam <thomas@xteddy.org>
Date:   Wed Aug 29 09:04:51 2012 +0100

    Initialise event timer before using it.

diff --git a/tty-keys.c b/tty-keys.c
index 5caf516..fe0e084 100644
--- a/tty-keys.c
+++ b/tty-keys.c
@@ -27,6 +27,9 @@
 
 #include "tmux.h"
 
+#define EV_TIMER_PENDING(t) (evtimer_initialized(t) && \
+		evtimer_pending(t, NULL))
+
 /*
  * Handle keys input from the outside terminal. tty_keys[] is a base table of
  * supported keys which are looked up in terminfo(5) and translated into a
@@ -533,7 +536,7 @@ partial_key:
 	 * timer has expired, give up waiting and send the escape.
 	 */
 	if ((tty->flags & TTY_ESCAPE) &&
-	    !evtimer_pending(&tty->key_timer, NULL)) {
+	    !EV_TIMER_PENDING(&tty->key_timer)) {
 		evbuffer_drain(tty->event->input, 1);
 		key = '\033';
 		goto handle_key;
@@ -543,7 +546,7 @@ partial_key:
 
 start_timer:
 	/* If already waiting for timer, do nothing. */
-	if (evtimer_pending(&tty->key_timer, NULL))
+	if (EV_TIMER_PENDING(&tty->key_timer))
 		return (0);
 
 	/* Start the timer and wait for expiry or more data. */
